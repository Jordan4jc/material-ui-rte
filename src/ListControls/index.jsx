import React from 'react'
import { makeStyles } from "@material-ui/core/styles"
import Button from '@material-ui/core/Button'
import ButtonGroup from '@material-ui/core/ButtonGroup'
import FormatListBulletedIcon from "@material-ui/icons/FormatListBulleted"
import FormatListNumberedIcon from "@material-ui/icons/FormatListNumbered"
import clsx from 'clsx';
import styles from "./styles"

const useStyles = makeStyles(styles, {
  name: "MuiRte-InlineListControls"
});

const LIST_STYLES = [
  {icon: FormatListNumberedIcon, style: 'ordered-list-item'},
  {icon: FormatListBulletedIcon, style: 'unordered-list-item'}
];

const ListControls = (props) => {
  const classes = useStyles()
  const currentStyle = props.editorState.getCurrentInlineStyle();
  const renderIcon = type => {
    let IconName = type.icon
    return <IconName className={clsx({
      [classes.activeIcon]: currentStyle.has(type.style)
    })} />
  }
  return (
    <ButtonGroup size="small" variant="contained">
      {LIST_STYLES.map((type) =>
        <Button key={type.style} onClick={() => props.onChange(type.style)}>{renderIcon(type)}</Button>
      )}
    </ButtonGroup>
  );
};

export default ListControls