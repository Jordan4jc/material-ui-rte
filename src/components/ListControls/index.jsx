import React from "react"
// import { makeStyles } from "@material-ui/core/styles"
import Button from "@material-ui/core/Button"
import ButtonGroup from "@material-ui/core/ButtonGroup"
import FormatListBulletedIcon from "@material-ui/icons/FormatListBulleted"
import FormatListNumberedIcon from "@material-ui/icons/FormatListNumbered"
import clsx from "clsx"
// import styles from "./styles"

// const useStyles = makeStyles(styles, {
//   name: "MuiRte-InlineListControls"
// })

const LIST_STYLES = [
  { icon: FormatListNumberedIcon, style: "ordered-list-item" },
  { icon: FormatListBulletedIcon, style: "unordered-list-item" }
]

const ListControls = props => {
  // const classes = useStyles()
  const classes = {}
  const currentStyle = props.editorState.getCurrentInlineStyle()
  const renderIcon = type => {
    let IconName = type.icon
    return (
      <IconName
        className={clsx({
          [classes.activeIcon]: currentStyle.has(type.style)
        })}
      />
    )
  }
  function listStyles() {
    if (Array.isArray(props.controls)) {
      return LIST_STYLES.filter(
        b => props.controls.indexOf(b.style.replace("-item", "")) > -1
      )
    } else {
      return LIST_STYLES
    }
  }
  return (
    <ButtonGroup size="small" variant="contained">
      {listStyles().map(type => (
        <Button key={type.style} onClick={() => props.onChange(type.style)}>
          {renderIcon(type)}
        </Button>
      ))}
    </ButtonGroup>
  )
}

export default ListControls
