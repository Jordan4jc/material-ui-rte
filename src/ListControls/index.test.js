import React from "react"
import { shallow } from "enzyme"
import { EditorState } from "draft-js"
import { unwrap } from "@material-ui/core/test-utils";
import ListControls from "./index"

const ListControlsUnwrapped = unwrap(ListControls);

describe("ListControls", () => {
  let editorState

  beforeEach(() => {
    editorState= EditorState.createEmpty()
  })

  it("Should render list control", () => {
    const component = shallow(<ListControlsUnwrapped editorState={editorState} classes={{}} />);
  
    expect(component).toMatchSnapshot();
  });
});