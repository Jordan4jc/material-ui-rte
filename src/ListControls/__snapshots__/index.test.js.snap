// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListControls Should render list control 1`] = `
<WithStyles(ForwardRef(ButtonGroup))
  size="small"
  variant="contained"
>
  <WithStyles(ForwardRef(Button))
    key="ordered-list-item"
    onClick={[Function]}
  >
    <FormatListNumberedIcon
      className=""
    />
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Button))
    key="unordered-list-item"
    onClick={[Function]}
  >
    <FormatListBulletedIcon
      className=""
    />
  </WithStyles(ForwardRef(Button))>
</WithStyles(ForwardRef(ButtonGroup))>
`;
